lib/travis/build/appliances/restart_mysql.rb

```diff
@@ -0,0 +1,14 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class RestartMysql < Base
+        def apply
+          sh.cmd '(ls /var/run/mysqld/mysqld.sock >& /dev/null) || sudo service mysql restart'
```

Comment:
Can this be `test -e /var/run/mysqld/mysqld.sock` instead of `ls`?  Using `ls` for noninteractive stuff is a recipe for sadness, IMHO.

/cc @tianon